src/Uno.UI/UI/Xaml/Internal/RootVisual.cs

```diff
@@ -98,5 +103,52 @@ protected override Size ArrangeOverride(Size finalSize)
 
 			return finalSize;
 		}
+
+
+#if HAS_UNO
+		// Uno specific: To ensure focus is properly lost when clicking "outside" app's content,
+		// we set focus here. In case UWP, focus is set to the root ScrollViewer instead,
+		// but Uno does not have it on all targets yet.
+		private bool _isLeftButtonPressed = false;
+
+		private void RootVisual_PointerPressed(object sender, Windows.UI.Xaml.Input.PointerRoutedEventArgs e)
+		{
+			if (e.Handled)
+			{
+				return;
+			}
+
+			var point = e.GetCurrentPoint(this);
+			_isLeftButtonPressed = point.Properties.IsLeftButtonPressed;
```

Comment:
FYI we have an internal helper for that https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/UIElement.Pointers.cs#L1056 ... but this won't let you have a distinction between left or right button ... which BTW is weak, cf. the comment above ;)